src/rattler_build_conda_compat/modify_recipe.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+import hashlib
+import io
+import re
+from typing import TYPE_CHECKING, Any, Generator
+
+import requests
+from ruamel.yaml import YAML
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+yaml = YAML()
+yaml.preserve_quotes = True
+
+
+def _update_build_number_in_context(recipe: dict[str, Any], new_build_number: int) -> bool:
+    for key in recipe.get("context", {}):
+        if key.startswith("build_") or key == "build":
+            recipe["context"][key] = new_build_number
+            return True
+    return False
+
+
+def _update_build_number_in_recipe(recipe: dict[str, Any], new_build_number: int) -> bool:
+    is_modified = False
+    if "build" in recipe and "number" in recipe["build"]:
+        recipe["build"]["number"] = new_build_number
+        is_modified = True
+
+    if "outputs" in recipe:
+        for output in recipe["outputs"]:
+            if "build" in output and "number" in output["build"]:
+                output["build"]["number"] = new_build_number
+                is_modified = True
+
+    return is_modified
+
+
+def update_build_number(file: Path, new_build_number: int) -> str:
+    # This function should be called to update the build number of the recipe
+    # in the meta.yaml file.
+    with file.open("r") as f:
+        data = yaml.load(f)
+    build_number_modified = _update_build_number_in_context(data, new_build_number)
+    if not build_number_modified:
+        build_number_modified = _update_build_number_in_recipe(data, new_build_number)
```

Comment:
I think here we can just call 
`_update_build_number_in_recipe(data, new_build_number)`